runtime/shiden/src/lib.rs

```diff
@@ -802,6 +802,83 @@ impl pallet_xc_asset_config::Config for Runtime {
     type WeightInfo = weights::pallet_xc_asset_config::WeightInfo<Self>;
 }
 
+/// The type used to represent the kinds of proxying allowed.
+#[derive(
+    Copy,
+    Clone,
+    Eq,
+    PartialEq,
+    Ord,
+    PartialOrd,
+    Encode,
+    Decode,
+    RuntimeDebug,
+    MaxEncodedLen,
+    scale_info::TypeInfo,
+)]
+pub enum ProxyType {
+    CancelProxy,
+    DappsStaking,
+}
+
+impl Default for ProxyType {
+    fn default() -> Self {
+        Self::CancelProxy
+    }
+}
+
+impl InstanceFilter<RuntimeCall> for ProxyType {
+    fn filter(&self, c: &RuntimeCall) -> bool {
+        match self {
+            ProxyType::CancelProxy => {
+                matches!(
+                    c,
+                    RuntimeCall::Proxy(pallet_proxy::Call::reject_announcement { .. })
+                )
+            }
+            ProxyType::DappsStaking => {
+                matches!(c, RuntimeCall::DappsStaking(..) | RuntimeCall::Utility(..))
```

Comment:
I believe Utility makes possible to do any call using batch meta-calls, lest exclude it this time.

Reply:
Utility call is necessary for oak integration as the compounding call in oak is a utility call

Reply:
@akru the call filter is bypassed only if origin is `root`. So it's not possible to bypass the filter using _batch meta call_.